src/_guides/libraries/library-tour.md

```diff
@@ -1293,6 +1293,64 @@ await Future.wait([
 print('Done with all the long steps!');
 ```
 
+`Future.wait()` returns a future which will complete once all the provided
+futures have completed, either with their results, or with an error if any of
+the provided futures fail. 
+
+#### Handling errors for multiple futures
+
+You can also wait for parallel operations on an [iterable]({{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-async/FutureIterable/wait.html)
+or [record]({{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-async/FutureRecord2/wait.html)
```

Comment:
```suggestion
of [record]({{site.dart-api}}/{{site.data.pkg-vers.SDK.channel}}/dart-async/FutureRecord2/wait.html)
```

Right?

Reply:
Records aren't actually part of the iterable family. So these future APIs for iterables and records are separate. 

Kind of unrelated, but the implementation distinction between records' implementation / API and the rest of the collections API has made it hard to generalize all them with a single word (e.g. "you can aggregate different values in a single object using Dart's collection types (lists, maps, sets) ... _and_ records"). I wish I could just call them a collection.

Reply:
The word I'd use is that they're *composite* objects (can contain other objects).
More precisely "composite value containers" (even if it's a bit redundant).

Most objects are composite in some sense, but some objects has that as their primary objective, being exactly the sum of their parts with no extra semantic interpretation or functionality. A list is just a list. A task queue might be a list, but it has a role. It's not _just_ a data container.

"Collections" are traditionally *mutable* homogeneous composite objects, whose primary task is to collect and access their values. Very traditionally they are lists, sets, queues and/or maps (maps are sometimes considered a collection, other times not, qv., `java.util.Collection` and `java.util.Map`). Good old "data structures".

"Structs", "records" or "tuples" are heterogeneous composite objects, but still mostly without extra functionality on top of just carrying the data values. Aka. "data objects"/"data classes".

Don't know if that helps at all :grin:

Reply:
Composite! That's exactly what I was looking for. I've weaseled my way around lumping records in with the rest of the collections so far, so I don't think I need to go back and correct anything, but if the need arises again (and I'm sure it will) I'll use "composite". Might even be a good entry for the new glossary. Thanks!